File: easy/3880_minimum_absolute_difference_between_two_values/Solution.java
class Solution {
  public int minAbsoluteDifference(int[] nums) {
    int minDifference = Integer.MAX_VALUE;

    int prevNum = 0;
    int prevI = 0;

    for (int i = 0; i < nums.length; ++i) {
      if (nums[i] == 0) {
        continue;
      }

      if (prevNum == 3 - nums[i] && i - prevI < minDifference) {
        minDifference = i - prevI;
      }

      prevNum = nums[i];
      prevI = i;
    }

    if (minDifference == Integer.MAX_VALUE) {
      return -1;
    } else {
      return minDifference;
    }
  }
}

File: easy/3880_minimum_absolute_difference_between_two_values/solution.py
from typing import List


class Solution:
    def minAbsoluteDifference(self, nums: List[int]) -> int:
        min_difference = len(nums) + 1

        for i in range(len(nums)):
            if nums[i] == 0:
                continue
            elif nums[i] == 1:
                other = 2
            else:
                other = 1

            for j in range(i + 1, len(nums)):
                if nums[j] == other:
                    min_difference = min(min_difference, abs(i - j))

        return min_difference if min_difference != len(nums) + 1 else -1
